Sources/SQLiteNIO/SQLiteData.swift

```diff
@@ -89,3 +108,37 @@ public enum SQLiteData: Equatable, Encodable, CustomStringConvertible {
         }
     }
 }
+
+extension SQLiteData {
+	init(sqliteValue: OpaquePointer) throws {
+		switch sqlite_nio_sqlite3_value_type(sqliteValue) {
+		case SQLITE_NULL:
+			self = .null
+		case SQLITE_INTEGER:
+			self = .integer(Int(sqlite_nio_sqlite3_value_int64(sqliteValue)))
+		case SQLITE_FLOAT:
+			self = .float(sqlite_nio_sqlite3_value_double(sqliteValue))
+		case SQLITE_TEXT:
+			self = .text(String(cString: sqlite_nio_sqlite3_value_text(sqliteValue)!))
+		case SQLITE_BLOB:
+			if let bytes = sqlite_nio_sqlite3_value_blob(sqliteValue) {
+				let count = Int(sqlite_nio_sqlite3_value_bytes(sqliteValue))
+				var buffer = ByteBufferAllocator().buffer(capacity: count)
+				buffer.writeBytes(UnsafeBufferPointer(
+					start: bytes.assumingMemoryBound(to: UInt8.self),
+					count: count
+				))
+
```

Comment:
There's a lot happening in the blob case, but we can make it simpler by using some newer NIO features (and omitting the unnecessary pointer binding):

```swift
if let bytes = sqlite_nio_sqlite3_value_blob(sqliteValue) {
    let count = Int(sqlite_nio_sqlite3_value_bytes(sqliteValue))
    let buffer = ByteBuffer(bytes: UnsafeRawBufferPointer(start: bytes, count: count))
    self = .blob(buffer)
}
```